tensorflow/python/ops/image_ops.py

```diff
@@ -1388,6 +1391,66 @@ def _crop_and_resize_shape(op):
 ops.RegisterShape('NonMaxSuppression')(common_shapes.call_cpp_shape_fn)
 
 
+def decode_image(contents, channels=None, ratio=None, fancy_upscaling=None,
+                 try_recover_truncated=None, acceptable_fraction=None,
+                 name=None):
+  """Convenience function for `decode_gif`, `decode_jpeg`, and `decode_png`.
+  Detects whether an image is a GIF, JPEG, or PNG, and performs the appropriate 
+  operation to convert the input bytes `string` into a `Tensor` of type `uint8`.
+
+  Note: `decode_gif` returns a 4-D array `[num_frames, height, width, 3]`, as 
+  opposed to `decode_jpeg` and `decode_png`, which return 3-D arrays 
+  `[height, width, num_channels]`. Make sure to take this into account when 
+  constructing your graph if you are intermixing GIF files with JPEG and/or PNG 
+  files.
+
+  Args:
+    contents: 0-D `string`. The encoded image bytes.
+    channels: Number of color channels for the decoded image.
+    ratio: Downscaling ratio (only used when decoding JPEG images)
+    fancy_upscaling: If true use a slower but nicer upscaling of the chroma 
+      planes (yuv420/422 JPEG images only).
+    try_recover_truncated: If true, try to recover an image from truncated input
+      (only used when decoding JPEG images).
+    acceptable_fraction: The minimum required fraction of lines before a 
+      truncated input is accepted (only used when decoding JPEG images).
+  
+  Returns:
+    `Tensor` with type `uint8`. Shape `[height, width, num_channels]` for JPEG 
+      and PNG images. Shape `[num_frames, height, width, 3]` for GIF images.
+  """
+  with ops.name_scope(name, 'decode_image') as scope:
+    def _gif():
+      return gen_image_ops.decode_gif(contents)
```

Comment:
I don't think so? I'm digging around the[ `decode` implementation in tensorflow/core/lib/gif](https://github.com/tensorflow/tensorflow/blob/6b1d4fd8090d44d20fdadabf06f1a9b178c3d80c/tensorflow/core/lib/gif/gif_io.cc#L35), which currently has the number of channels hard coded to three. If you try to pass in a `channels` argument to `decode_gif`, you get an error, as [`channels` isn't registered as a parameter](https://github.com/tensorflow/tensorflow/blob/816ecb7a342c25c19daa8b19627346e1fb38e56f/tensorflow/core/ops/image_ops.cc#L522-L524). Additionally, the [`DecodeGifOp` only reads one value (the image contents) from the context](https://github.com/tensorflow/tensorflow/blob/6b1d4fd8090d44d20fdadabf06f1a9b178c3d80c/tensorflow/core/kernels/decode_gif_op.cc#L36).